pkg/webhook/webhook.go

```diff
@@ -270,7 +281,18 @@ func parsePodNetworkSelections(podNetworks, defaultNamespace string) ([]*multus.
 	/* fill missing namespaces with default value */
 	for _, networkSelection := range networkSelections {
 		if networkSelection.Namespace == "" {
-			networkSelection.Namespace = defaultNamespace
+			if defaultNamespace == "" {
+				// Ignore the AdmissionReview request when the following conditions are met:
+				// 1) net-attach-def annotation doesn't contain a valid namespace
+				// 2) defaultNamespace retrieved from admission request is empty
+				// Pod admission would fail in subsquent call "getNetworkAttachmentDefinition"
+				// if no namespace is specified. We don't want to fail the pod creation
+				// in such case since it is possible that pod is not a SR-IOV pod
+				glog.Warningf("The admission request doesn't contain a valid namespace, ignoring...")
+				return nil, nil
```

Comment:
shouldn't you return empty NetworkSelectionElement array here instead of nil ?

Reply:
I think both nil or empty slice would work the same, looking at golang doc, it says:
```
In practice, nil slices and empty slices can often be treated in the same way:

    they have zero length and capacity,
    they can be used with the same effect in for loops and append functions,
    and they even look the same when printed.
```

In our case, we use len() to query the length of slice and use for loop against it.
I can change it if you prefer to.

Reply:
ah! ok :), then let's just use nil as it is.